src/Compilers/Core/Portable/InternalUtilities/FatalError.cs

```diff
@@ -121,37 +141,98 @@ public static bool ReportAndPropagateUnlessCanceled(Exception exception)
         /// <see cref="CancellationToken.IsCancellationRequested"/> set if cancellation is expected.</param>
         /// <returns><see langword="false"/> to avoid catching the exception.</returns>
         [DebuggerHidden]
-        public static bool ReportAndPropagateUnlessCanceled(Exception exception, CancellationToken contextCancellationToken)
+        public static bool ReportAndPropagateUnlessCanceled(Exception exception, CancellationToken contextCancellationToken, ErrorSeverity severity = ErrorSeverity.Uncategorized)
         {
             if (IsCurrentOperationBeingCancelled(exception, contextCancellationToken))
             {
                 return false;
             }
 
-            return ReportAndPropagate(exception);
+            return ReportAndPropagate(exception, severity);
         }
 
-#if !NET20
+        /// <summary>
+        /// Report an error.
+        /// Calls <see cref="Handler"/> and doesn't pass the exception through (the method returns true).
+        /// This is generally expected to be used within an exception filter as that allows us to
+        /// capture data at the point the exception is thrown rather than when it is handled.
+        /// However, it can also be used outside of an exception filter. If the exception has not
+        /// already been thrown the method will throw and catch it itself to ensure we get a useful
+        /// stack trace.
+        /// </summary>
+        /// <returns>True to catch the exception.</returns>
+        [DebuggerHidden]
+#if COMPILERCORE
+        private
+#else
+        public
+#endif
+            static bool ReportAndCatch(Exception exception, ErrorSeverity severity = ErrorSeverity.Uncategorized)
+        {
+            Report(exception, severity);
+            return true;
+        }
+
+        [DebuggerHidden]
+#if COMPILERCORE
+        private
+#else
+        public
+#endif
+            static bool ReportWithDumpAndCatch(Exception exception, ErrorSeverity severity = ErrorSeverity.Uncategorized)
```

Comment:
Where do we use this?

Reply:
I guess the one use got removed, so we could get rid of this.

Reply:
Oh right: I wanted to keep it around until we have confidence that the dump telemetry requests are working again, as this is the backup until they are.